src/Aquifer.Jobs/Managers/SendResourceAssignmentNotifications.cs

```diff
@@ -0,0 +1,101 @@
+using Aquifer.Common.Services;
+using Aquifer.Data;
+using Aquifer.Data.Enums;
+using Aquifer.Jobs.Configuration;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Aquifer.Jobs.Managers;
+
+public class SendResourceAssignmentNotifications(
+    IOptions<ConfigurationOptions> _configurationOptions,
+    AquiferDbContext _dbContext,
+    IEmailService _emailService,
+    ILogger<SendResourceAssignmentNotifications> _logger)
+{
+    private const string _everyTenMinutesCronSchedule = "0 */10 * * * *";
+    private const string _tenSecondDelayInterval = "00:00:10";
+
+    [Function(nameof(SendResourceAssignmentNotifications))]
+    [FixedDelayRetry(maxRetryCount: 1, delayInterval: _tenSecondDelayInterval)]
+#pragma warning disable IDE0060 // Remove unused parameter: A (non-discard) TimerInfo parameter is required for correct Azure bindings.
+    public async Task RunAsync([TimerTrigger(_everyTenMinutesCronSchedule)] TimerInfo timerInfo, CancellationToken ct)
+#pragma warning restore IDE0060 // Remove unused parameter
+    {
+        var jobHistory = await _dbContext.JobHistory
+            .AsTracking()
+            .SingleAsync(jh => jh.JobId == JobId.SendResourceAssignmentNotifications, ct);
+
+        // Possible Improvement: Only send notifications when a user is assigned to the *most recent* ResourceContentVersion.
+        var userHistories = await _dbContext.ResourceContentVersionAssignedUserHistory
+                .Where(rcvauh =>
+                    rcvauh.AssignedUser != null &&
+                    rcvauh.AssignedUser.AquiferNotificationsEnabled &&
+                    rcvauh.AssignedUserId == rcvauh.ResourceContentVersion.AssignedUserId &&
+                    (rcvauh.Created > jobHistory.LastProcessed))
+                .Select(rcvauh => new
+                {
+                    AssignedUser = rcvauh.AssignedUser!,
+                    rcvauh.Created,
+                    ParentResourceName = rcvauh.ResourceContentVersion.ResourceContent.Resource.ParentResource.DisplayName,
+                    rcvauh.ResourceContentVersion.ResourceContentId,
+                    ResourceName = rcvauh.ResourceContentVersion.ResourceContent.Resource.EnglishLabel,
+                })
+                .ToListAsync(ct);
+
+        var usersByIdMap = userHistories
+            .Select(uh => uh.AssignedUser)
+            .DistinctBy(u => u.Id)
+            .ToDictionary(au => au.Id);
+
+        var templatedEmails = userHistories
+            .GroupBy(uh => uh.AssignedUser.Id)
+            .Select(userGrouping => new TemplatedEmail(
+                From: NotificationsHelper.NotificationSenderEmailAddress,
+                // Template Designer: https://mc.sendgrid.com/dynamic-templates/d-d85f76c6b4d344f5bc8b90b27cc40cc3/version/b6955fec-6f2e-41f7-a9f5-fb695a5b8ed7/editor
+                TemplateId: "d-d85f76c6b4d344f5bc8b90b27cc40cc3",
```

Comment:
See https://biblionexusworkspace.slack.com/archives/C05K5N7F44Q/p1729881075224309 for design review of this email template.